app/src/main/java/org/mozilla/focus/browser/BrowserFragment.kt

```diff
@@ -165,20 +171,54 @@ class BrowserFragment : IWebViewLifecycleFragment() {
                 // so we need to do our own caching: see FocusedDOMElementCache for details.
                 if (!isVisible) { webView?.focusedDOMElement?.cache() }
             }
+            registerForContextMenu(browserOverlay.tileContainer)
         }
 
         layout.progressBar.initialize(this)
 
         return layout
     }
 
+    override fun onContextItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.remove -> {
+                val homeTileAdapter = tileContainer.adapter as HomeTileAdapter
+                val tileToRemove = homeTileAdapter.getItemAtPosition(browserOverlay.getFocusedTilePosition())
+                        ?: return false
+                // This assumes that since we're deleting from a Home Tile object that we created
+                // that the Uri is valid, so we do not do error handling here.
+                when (tileToRemove) {
+                    is BundledHomeTile -> {
+                        val tileUri = tileToRemove.url.toUri()
+                        if (tileUri != null) {
+                            BundledTilesManager.getInstance(context).unpinSite(context, tileUri)
+                            homeTileAdapter.removeItemAtPosition(browserOverlay.getFocusedTilePosition())
+                        }
+                    }
+                    is CustomHomeTile -> {
+                        CustomTilesManager.getInstance(context).unpinSite(context, tileToRemove.url)
+                        homeTileAdapter.removeItemAtPosition(browserOverlay.getFocusedTilePosition())
+                    }
+                }
+                TelemetryWrapper.homeTileRemovedEvent(tileToRemove)
+                return true
+            }
+        }
+        return false
```

Comment:
You should try to make your `when` exhaustive, even if it has to have an else case (e.g. for ints like these):
```kotlin
return when (item.itemId) {
    R.id.remove -> {
        ...
        true
    }
    else -> false
}